scripts/generator/graphql_generator/csharp/SDK/iOS/PaymentDeserializer.cs.erb

```diff
@@ -36,16 +37,26 @@ namespace <%= namespace %>.SDK.iOS {
             var shippingContactDictionary = (Dictionary<string, object>)dictionary[PaymentField.ShippingContact.ToString("G")];
             ShippingAddress = new MailingAddressInput(shippingContactDictionary);
 
+            // Email
+            Email = (string)shippingContactDictionary["email"];
+
             // Shipping Identifier
-            ShippingIdentifier = (string)dictionary[PaymentField.ShippingIdentifier.ToString("G")];
+            if (dictionary.ContainsKey(PaymentField.ShippingIdentifier.ToString("G"))) {
+                ShippingIdentifier = (string)dictionary[PaymentField.ShippingIdentifier.ToString("G")];
+            }
 
             // Transaction Identifier
             var tokenDictionary = (Dictionary<string, object>)dictionary[PaymentField.TokenData.ToString("G")];
             TransactionIdentifier = (string)tokenDictionary[TokenField.TransactionIdentifier.ToString("G")];
 
             // Payment Data
-            var paymentDictionary = tokenDictionary[TokenField.PaymentData.ToString("G")];
-            PaymentData = Json.Serialize(paymentDictionary);
+            // Payment Data can be null when running on a simulator
+            if (tokenDictionary.ContainsKey(TokenField.PaymentData.ToString("G"))) {
```

Comment:
Would it make sense to use `#if IOS_SIMULATOR_SDK` here to blank out the `PaymentData`? It would allow us to differentiate between running on the simulator and an issue with the tokenDictionary. This also might not even be an issue 🤷‍♂️